pkg/querier/blocks_store_queryable_test.go

```diff
@@ -763,81 +794,159 @@ func TestBlocksStoreQuerier_Select(t *testing.T) {
-			if testData.queryShardID != "" {
-				matchers = append(matchers, labels.MustNewMatcher(labels.MatchEqual, sharding.ShardLabel, testData.queryShardID))
-			}
+					stores := &blocksStoreSetMock{mockedResponses: storeSetResponses}
+					finder := &blocksFinderMock{}
+					finder.On("GetBlocks", mock.Anything, "user-1", minT, maxT).Return(testData.finderResult, map[ulid.ULID]*bucketindex.BlockDeletionMark(nil), testData.finderErr)
+
+					ctx := limiter.AddQueryLimiterToContext(context.Background(), testData.queryLimiter)
+					st, ctx := stats.ContextWithEmptyStats(ctx)
+					q := &blocksStoreQuerier{
+						ctx:         ctx,
+						minT:        minT,
+						maxT:        maxT,
+						userID:      "user-1",
+						finder:      finder,
+						stores:      stores,
+						consistency: NewBlocksConsistencyChecker(0, 0, log.NewNopLogger(), nil),
+						logger:      log.NewNopLogger(),
+						metrics:     newBlocksStoreQueryableMetrics(reg),
+						limits:      testData.limits,
+					}
 
-			sp := &storage.SelectHints{Start: minT, End: maxT}
-			set := q.Select(true, sp, matchers...)
-			if testData.expectedErr != nil {
-				assert.ErrorContains(t, set.Err(), testData.expectedErr.Error())
-				assert.IsType(t, set.Err(), testData.expectedErr)
-				assert.False(t, set.Next())
-				assert.Nil(t, set.Warnings())
-				return
-			}
+					matchers := []*labels.Matcher{
+						labels.MustNewMatcher(labels.MatchEqual, labels.MetricName, metricName),
+					}
+					if testData.queryShardID != "" {
+						matchers = append(matchers, labels.MustNewMatcher(labels.MatchEqual, sharding.ShardLabel, testData.queryShardID))
+					}
 
-			require.NoError(t, set.Err())
-			assert.Len(t, set.Warnings(), 0)
-
-			// Read all returned series and their values.
-			var actualSeries []seriesResult
-			var it chunkenc.Iterator
-			for set.Next() {
-				var actualValues []valueResult
-
-				it = set.At().Iterator(it)
-				for valType := it.Next(); valType != chunkenc.ValNone; valType = it.Next() {
-					assert.Equal(t, valType, chunkenc.ValFloat)
-					t, v := it.At()
-					actualValues = append(actualValues, valueResult{
-						t: t,
-						v: v,
-					})
-				}
+					sp := &storage.SelectHints{Start: minT, End: maxT}
+					set := q.Select(true, sp, matchers...)
+					if testData.expectedErr != nil {
+						if streaming && set.Err() == nil {
+							// In case of streaming, the error can happen during iteration.
+							foundErr := false
+							for set.Next() {
+								it := set.At().Iterator(nil)
+								for it.Next() != chunkenc.ValNone { // nolint
+								}
+								err := it.Err()
+								if err != nil {
+									assert.ErrorContains(t, err, testData.expectedErr.Error())
+									assert.ErrorIs(t, err, testData.expectedErr)
+									foundErr = true
+									break
+								}
+							}
+							assert.True(t, foundErr)
+						} else {
+							assert.ErrorContains(t, set.Err(), testData.expectedErr.Error())
+							assert.IsType(t, set.Err(), testData.expectedErr)
```

Comment:
Same as above - can we simplify this to just `assert.ErrorIs` here?

Reply:
For some reasons `assert.ErrorIs` fails here though in the test failure message the expected and actual error message is the exact same!

Reply:
What if we use `assert.Equal()`?

Reply:
assert.Equal() fails because it is not exactly equal. I am thinking to leave this one as it is.

Reply:
Yeah that's fine, not the end of the world.